public/app/types/unified-alerting.ts

```diff
@@ -50,6 +50,11 @@ export interface AlertingRule extends RuleBase {
   };
   state: PromAlertingRuleState;
   type: PromRuleType.Alerting;
+
+  /**
+   * Pending period in seconds, aka for. 0 or undefined means no pending period
+   */
+  duration?: number;
```

Comment:
why do we need to add this filed in here in this PR? 🤔

Reply:
This is the field we use to get the pending period for vanilla prometheus rules. https://github.com/grafana/grafana/pull/92728/files#diff-88cbd9c2cc465584ea3cd717de616b88cec04a879be7b46a75ca8b3a2bc1ecd1R19-R20
The API has been returning it for a long time, just our types weren't covering it